File: DesignPatternsDotNet/Composite/BookType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Collections;

namespace Composite
{
    public class BookType : BookComponent
    {
        ArrayList books = new ArrayList();
        string typeName;
        string typeDescription;
        public string TypeName
        {
            get
            {
                return typeName;
            }
            set { typeName = value; }
        }
        public string TypeDescription
        {
            get{
            return typeDescription;}
            set{typeDescription = value;
            }
        }
        public BookType(string name, string des)
        {
            typeName = name;
            typeDescription = des;
        }
        public override void Add(BookComponent aComponent)
        {
            books.Add(aComponent);
        }
        public override void Delete(BookComponent aComponent)
        {
            books.Remove(aComponent);
        }
        public override BookComponent GetComponent(int index)
        {
            return (BookComponent)books[index];
        }
        public override void DisplayBookInfo()
        {
		Console.WriteLine(TypeName+" "+TypeDescription);
        foreach (var i in books)
            {
            BookComponent aBook = (BookComponent)i;
            aBook.DisplayBookInfo();
            }
		
	    }
    }
}

File: DesignPatternsJava/src/main/java/com/Composite/BookType.java
package com.Composite;

import java.util.ArrayList;
import java.util.Iterator;

public class BookType extends BookComponent{
	ArrayList books = new ArrayList();
	String typeName; 
	String typeDescription;	
	public String getTypeName() {
		return typeName;
	}
	public void setTypeName(String typeName) {
		this.typeName = typeName;
	}
	public String getTypeDescription() {
		return typeDescription;
	}
	public void setTypeDescription(String description) {
		this.typeDescription = description;
	}
	public BookType(String name, String des){
		typeName=name;
		typeDescription=des;
	}
	public void add(BookComponent aComponent) {
		books.add(aComponent);		
	}
	public void delete(BookComponent aComponent) {
		books.remove(aComponent);		
	}
	public BookComponent getComponent(int index){
		return (BookComponent)books.get(index);
	}
	public void displayBookInfo(){
		System.out.println(getTypeName()+" "+getTypeDescription());
		Iterator i = books.iterator();
		while(i.hasNext()){
			((BookComponent)i.next()).displayBookInfo();
		}		
	}
}
